RaftRs/consensus/RaftRs.tla

```diff
@@ -0,0 +1,1232 @@
+            THEN LogCount(log'[follower]) <= LogCount(log'[leader])
+            ELSE TRUE
+       ELSE TRUE
+
+CommitIdxLELogLen ==
+    \A i \in Servers: commitIndex'[i] <= LogCount(log'[i])
+
+LeaderCommitCurrentTermLogs ==
+    \A i \in Servers:
+        IF raftState'[i] = Leader
+        THEN IF commitIndex[i] /= commitIndex'[i]
+             THEN log'[i][commitIndex'[i]].term = currentTerm'[i]
+             ELSE TRUE
+        ELSE TRUE
+
+NewLeaderTermNotInLog ==
+    \A i \in Servers:
+        IF raftState'[i] = Leader /\ raftState[i] /= Leader
+        THEN \A j \in Servers \ {i}:
+                \A n \in DOMAIN log'[j]:
+                    log'[j][n].term /= currentTerm'[i]
+        ELSE TRUE
+
+LeaderTermLogHasGreatestIdx ==
+    \A i \in Servers:
+        IF raftState'[i] = Leader
+        THEN \A j \in Servers \ {i}:
+                LET IncTermLogCount(a, b) == IF a.term = currentTerm'[i] THEN b + 1 ELSE b
+                IN FoldSeq(IncTermLogCount, 0, log'[i]) >= FoldSeq(IncTermLogCount, 0, log'[j])
+        ELSE TRUE
+
+CheckLeader ==
+    \A i \in Servers:
+       raftState[i] /= Leader
+
+InvSequence == <<
+    ElectionSafety,
+    LeaderAppendOnly,
+    LogMatching,
+    MonotonicCurrentTerm,
+    MonotonicCommitIdx,
+    MonotonicMatchIdx,
+    CommittedLogDurable,
+    CommittedLogReplicatedMajority,
+    NextIdxGtMatchIdx,
+    NextIdxGtZero,
+    \* CheckLeader
+    FollowerLogLELeaderLogAfterAE,
+    CommitIdxLELogLen,
+    LeaderCommitCurrentTermLogs,
+    NewLeaderTermNotInLog,
+    LeaderTermLogHasGreatestIdx
+
+>>
+
+INV == Len(SelectSeqWithIdx(inv, LAMBDA x, y: ~x /\ y \notin netman.no_inv)) = 0
+
+
+
+ (***************************************************************************
+  Next actions
+ ***************************************************************************)
+
+DoElectionTimeout ==
+    /\ PrePrune(netman.n_elec, "MaxElectionTimes")
+    /\ \E n \in Servers: CheckStateIs(n, Follower) /\ BecomeCandidate(n)
+    /\ inv' = InvSequence
+
+
+DoHeartBeat ==
+    /\ PrePrune(netman.n_hb, "MaxHeartBeatTimes")
+    /\ \E n \in Servers:
+        /\ raftState[n] = Leader
+        /\ SendHeartBeatAll(n)
+    /\ inv' = InvSequence
+
+
+_DoRecvM(type, func(_)) ==
+    /\ \E src, dst \in Servers:
+        /\ src /= dst
+        /\ LET m == NetGetMsg(src, dst)
+           IN /\ m /= Nil
+              /\ m.type = type
+              /\ func(m)
+    /\ inv' = InvSequence
+
+
+DoHandleMsgRV == /\ _DoRecvM(M_RV, HandleMsgRV)
+
+DoHandleMsgRVR == /\ _DoRecvM(M_RVR, HandleMsgRVR)
+                  
+DoHandleMsgAE == /\ _DoRecvM(M_AE, HandleMsgAE)
+
+DoHandleMsgAER == /\ _DoRecvM(M_AER, HandleMsgAER)
+                  
+DoHandleMsgHB == /\ _DoRecvM(M_HB, HandleMsgHB)
+
+DoHandleMsgHBR == /\ _DoRecvM(M_HBR, HandleMsgHBR)
+                  
+DoRecvEntry ==
+    /\ PrePrune(netman.n_op, "MaxClientOperationsTimes")
+    /\ \E n \in Servers, v \in Commands: RecvEntry(n, v)
+    /\ inv' = InvSequence
+
+\* DoNetworkDrop ==
```

Comment:
why comment out

Reply:
because this is for TCP, in UDP test will have the use of Drop, i'll clean it